### PR TITLE
chore: remove WalletConnect v2 code

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This removes the last workspace packages that will allow us to move to a traditional RN app repo structure.
Part of the work to fully move off of the legacy monorepo structure.

Note: in the future once v2 is stable and we need to support it, we can either publish the `walletconnect-v1` and `walletconnect-v2` wrapper packages to NPM. Or WalletConnect will hopefully have a simpler way to support both versions 🤞 

### Tested

Updated tests.

### How others should test

Connecting Dapps with WalletConnect works as before.

### Related issues

N/A

### Backwards compatibility

Technically backward incompatible, but v2 has been disabled for a while.

### CODE REVIEW

File: packages/mobile/src/walletConnect/screens/ActionRequest.tsx [hunk 1]
```diff
@@ -18,39 +18,27 @@ import {
...
-  acceptRequest as acceptRequestV2,
-  denyRequest as denyRequestV2,
-  showRequestDetails as showRequestDetailsV2,
-} from 'src/walletConnect/v2/actions'
-import { ERROR_TYPE as WalletConnectErrorType } from 'walletconnect-v2/utils'
 
 type Props = StackScreenProps<StackParamList, Screens.WalletConnectActionRequest>
 
 function showRequestDetails(params: Props['route']['params'], infoString: string) {
   switch (params.version) {
```
Comment (MuckT) [inline packages/mobile/src/walletConnect/screens/ActionRequest.tsx:25]: Can this switch be removed now?

**Review by MuckT — COMMENTED** [review]
Should [RootStateSchema.json](https://github.com/valora-inc/wallet/pull/2324/files#diff-752e1b9267812b1bd437419f4f3f3d40e483aa066712d5c258bf61f93bce97e0) have been deleted?

Also some notes about switch statements that might not be needed and bumping the schema version since this PR was made.

**Comment by jeanregisser** [comment]
Thanks!

> * Should [RootStateSchema.json](https://github.com/valora-inc/wallet/pull/2324/files#diff-752e1b9267812b1bd437419f4f3f3d40e483aa066712d5c258bf61f93bce97e0) have been deleted?

It's been updated.

> * Also some notes about switch statements  that might not be needed.

Yep removed.

> * Bumping the schema version to 48 since this PR was made.

Done

**Review by gnardini — APPROVED** [review]
🙌

**Review by MuckT — APPROVED** [review]
🚀 Awesome!

**Comment by MuckT** [comment]
@jeanregisser it looks like the schema might be on 50 now.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest Android Internal Release Build V1.32.0 & updated the status in Test rails.

- User is able to connect and disconnect various dapps for dapps screen.

**Test rail links:**
**Android 10:** https://valoraapp.testrail.io/index.php?/tests/view/33532

**Devices:** Google Pixel XL (10.0)
CC: @jeanregisser 
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest iOS testflight release build V: 1.32.0 and updated the status in Test rails.

**Test rail link:**
**iOS 15:** https://valoraapp.testrail.io/index.php?/tests/view/33517

**Device:** iPhone 13 mini(15.1.1)

**CC:** @jeanregisser 
Thanks..!!

**Pull request merged**